core/src/main/java/org/elasticsearch/common/xcontent/XContentBuilder.java

```diff
@@ -998,6 +998,10 @@ public void close() {
         }
     }
 
+    public boolean isClosed() {
+        return generator.isClosed();
+    }
+
     public XContentGenerator generator() {
```

Comment:
would it be enough to only add `isClosed` to XContentGenerator? and maybe add a javadoc?